### DISCUSSION TITLE
Eloquent Model to prefer PHP 8 types over DocBlock annotations

### DISCUSSION THREAD

**Discussion by kode-emperor** [laravel/framework#46610] [Ideas] [answered]
The Eloquent\Model is still lacking PHP 8 type support, ` Concerns::GuardsAttributes ` still uses DocBlock annotations which makes it hard for third-party code using PHP 8 types to build compatible extensions.

The following link shows the current implementation of `Concerns::GuardsAttributes`
https://github.com/laravel/framework/blob/10.x/src/Illuminate/Database/Eloquent/Concerns/GuardsAttributes.php , consider changing this implementation to use PHP 8 types.

**Comment by shaedrich** [comment, marked as answer]
This was added to Laravel 10 via #44545
